src/geometamaker/cli.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import os
+import sys
+
+import click
+from pydantic import ValidationError
+
+import geometamaker
+
+root_logger = logging.getLogger()
+root_logger.setLevel(logging.DEBUG)
+handler = logging.StreamHandler(sys.stdout)
+formatter = logging.Formatter(
+    fmt='%(asctime)s %(name)-18s %(levelname)-8s %(message)s',
+    datefmt='%m/%d/%Y %H:%M:%S ')
+handler.setFormatter(formatter)
+
+
+@click.command(
+    help='''Describe properties of a dataset given by FILEPATH and write this
+    metadata to a .yml sidecar file. Or if FILEPATH is a directory, describe
+    all datasets within.''',
+    short_help='Generate metadata for geospatial or tabular data, or zip archives.')
+@click.argument('filepath', type=click.Path(exists=True))
+@click.option('-r', '--recursive', is_flag=True, default=False,
+              help='if FILEPATH is a directory, describe files '
+                   'in all subdirectories')
+@click.option('-nw', '--no-write', is_flag=True, default=False,
+              help='Dump metadata to stdout instead of to a .yml file. '
+                   'This option is ignored if `filepath` is a directory')
+def describe(filepath, recursive, no_write):
+    if os.path.isdir(filepath):
+        geometamaker.describe_dir(
+            filepath, recursive=recursive)
+    else:
+        resource = geometamaker.describe(filepath)
+        if no_write:
+            click.echo(geometamaker.utils.yaml_dump(
+                resource.model_dump(exclude=['metadata_path'])))
+        else:
+            resource.write()
+
+
+def echo_validation_error(error, filepath):
+    summary = u'\u2715' + f' {filepath}: {error.error_count()} validation errors'
+    click.secho(summary, fg='bright_red')
+    for e in error.errors():
+        location = ', '.join(e['loc'])
+        msg_string = (f"    {e['msg']}. [input_value={e['input']}, "
+                      f"input_type={type(e['input']).__name__}]")
+        click.secho(location, bold=True)
+        click.secho(msg_string)
```

Comment:
This is a _much_ nicer way to format text strings than the old terminal control codes!